src/v/cloud_storage/partition_manifest.cc

```diff
@@ -491,14 +493,44 @@ bool partition_manifest::advance_start_offset(model::offset new_start_offset) {
             new_head_segment = std::next(new_head_segment);
         }
 
-        if (new_head_segment == _segments.end()) {
-            _start_offset = new_start_offset;
-        } else {
-            _start_offset = new_head_segment->second.base_offset;
+        model::offset advanced_start_offset
+          = new_head_segment == end() ? new_start_offset
+                                      : new_head_segment->second.base_offset;
+
+        if (previous_start_offset > advanced_start_offset) {
+            vlog(
+              cst_log.error,
+              "Previous start offset is greater than the new one: "
+              "previous_start_offset={}, computed_new_start_offset={}, "
+              "requested_new_start_offset={}",
+              previous_start_offset,
+              advanced_start_offset,
+              new_start_offset);
+            return false;
+        }
```

Comment:
Taking this branch clearly indicates a bug somewhere, which makes the ERROR log a great candidate for an alert. How can we set this up?